src/execution/physical_plan/plan_delim_join.cpp

```diff
@@ -7,14 +7,21 @@
 #include "duckdb/execution/physical_plan_generator.hpp"
 #include "duckdb/planner/expression/bound_reference_expression.hpp"
 
+#include "duckdb/execution/operator/scan/physical_column_data_scan.hpp"
+
 namespace duckdb {
 
-static void GatherDelimScans(const PhysicalOperator &op, vector<const_reference<PhysicalOperator>> &delim_scans) {
+static void GatherDelimScans(PhysicalOperator &op, vector<const_reference<PhysicalOperator>> &delim_scans,
+                             idx_t delim_index) {
 	if (op.type == PhysicalOperatorType::DELIM_SCAN) {
+		// This should be auto &scan = op.Cast<PhysicalColumnDataScan>();, however, Cast<> does
+		// not work for this case currently.
+		PhysicalColumnDataScan &scan = reinterpret_cast<PhysicalColumnDataScan &>(op);
```

Comment:
This is not as nice as it could and maybe should be. Essentially, I instantiate `Cast<PhysicalColumnDataScan>` manually. Problem is that this operator's type is `PhysicalOperatorType::COLUMN_DATA_SCAN` by default. This then trips the exception in the definition of `Cast` in `logical_operator.hpp`, because the acutal operator type is `PhysicalOperatorType::DELIM_SCAN`.

Reply:
We've fixed this elsewhere by setting the `PhysicalOperatorType` to `PhysicalOperatorType::INVALID` - that way the `.Cast` always succeeds. Perhaps you can include that change in this PR?

Reply:
I see! That works.